Restructure of events into three categories and QA improvements (#383)
HL7/fhircast-docs: input/pagecontent/2-3-Events.md

### Event name

> The event name defines the events. Most FHIRcast events conform to an extensible syntax based upon FHIR resources. 

Reviewer: ```suggestion
The event name defines the event. Most FHIRcast events conform to an extensible syntax based upon FHIR resources. 

```

`event-name ::= ( fhir-resource  ) '-' ( 'suffix' )`

> The `fhir-resource` indicates the anchor-type that is focus of the event, the `suffix` defines the type of event.

Reviewer: ```suggestion
The `fhir-resource` indicates the focus of the event; the `suffix` defines the type of event.

```

### Event types

The FHIRcast specification supports many different events. These events are defined in the [event catalog](3_Events.html). The events can be grouped in different types. The following sections define the characteristics of these different event-types.

#### Context-change events

FHIRcast context-change events that describe context changes SHALL conform to the following extensible syntax. Patterned after the SMART on FHIR scope syntax and expressed in EBNF notation, the FHIRcast syntax for context-change related events is:

`hub.events ::= ( fhir-resource  ) '-' ( 'open' | 'close' )`

Context change events will include the resource the context change relates to. Common FHIR resources are: Patient, Encounter, ImagingStudy and DiagnosticReport.

In the case the resource refers to other FHIR resources that represent there own context, these can be included as well. E.g. an [`Encounter-open`](3-4-encounter-open.html) also refers to the patient that is the subject of the Encounter. What resources to include is defined in the corresponding event definition in the [event catalog](3_Events.html).

> FHIRcast supports all events that follow this format. For the most common events definitions have been provided in the [event catalog](3_Events.html).

Reviewer: ```suggestion
FHIRcast supports all events that follow this format. The most common events definitions have been provided in the [event catalog](3_Events.html).

```

#### Context-change events

FHIRcast context-change events that describe context changes SHALL conform to the following extensible syntax. Patterned after the SMART on FHIR scope syntax and expressed in EBNF notation, the FHIRcast syntax for context-change related events is:

`hub.events ::= ( fhir-resource  ) '-' ( 'open' | 'close' )`

Context change events will include the resource the context change relates to. Common FHIR resources are: Patient, Encounter, ImagingStudy and DiagnosticReport.

In the case the resource refers to other FHIR resources that represent there own context, these can be included as well. E.g. an [`Encounter-open`](3-4-encounter-open.html) also refers to the patient that is the subject of the Encounter. What resources to include is defined in the corresponding event definition in the [event catalog](3_Events.html).

FHIRcast supports all events that follow this format. For the most common events definitions have been provided in the [event catalog](3_Events.html).

#### Infrastructure events

> This event category contains events required to maintain the FHIRcast network. The main events in this category are:

Reviewer: ```suggestion
This event category contains events required to maintain a FHIRcast session. The main events in this category are:

```

Context change events will include the resource the context change relates to. Common FHIR resources are: Patient, Encounter, ImagingStudy and DiagnosticReport.

In the case the resource refers to other FHIR resources that represent there own context, these can be included as well. E.g. an [`Encounter-open`](3-4-encounter-open.html) also refers to the patient that is the subject of the Encounter. What resources to include is defined in the corresponding event definition in the [event catalog](3_Events.html).

FHIRcast supports all events that follow this format. The most common events definitions have been provided in the [event catalog](3_Events.html).

#### Infrastructure events

This event category contains events required to maintain the FHIRcast network. The main events in this category are:

| [`syncerror`](3-2-syncerror.html) | indicates refusal to follow context or inability to deliver an event
| [`heartbeat`](3-2-heartbeat.html) | for monitoring the connection to the hub

#### Selection events

Selection events use the suffix `select`. The format of selection events is:

`hub.events ::= ( fhir-resource  ) '-' ( select )`

> The `fhir-resource` indicates the context of the selection. The context of a select event typically contains two fields. One with the name of the fhir-resource holding a the anchor resource. One or more named `select` indicating the resources that selected.

Reviewer: ```suggestion
The `fhir-resource` indicates the context of the selection. The `context` element in a select event typically contains two fields. One with the name of the `fhir-resource` holding a the anchor resource and one or more named `select` indicating the resources that is selected.

```
